### DISCUSSION TITLE
Differences to undetected-chromedriver?

### DISCUSSION THREAD

**Discussion by alexreg** [kaliiiiiiiiii/Selenium-Profiles#19] [Q&A] [answered]
Hey. This looks like a great project and one you've put a lot of work into... As a current user of undetected-chromedriver, I was just wondering if you might be able to summarise what pluses/minuses this library has vs undetected-chromedriver when it comes to detection by anti-bot mechanisms. Obviously, your library provides other features when it comes to device profiles and a few other things, though I'm mainly interested in the anti-detection capabilities for now.

**Comment by ttlns** [comment, marked as answer]
### Undetectability (headfull)
The most of it is __similar__ like in undetected-chromediver (stuff in [/scripts/undetected.py](https://github.com/kaliiiiiiiiii/Selenium-Profiles/blob/master/src/selenium_profiles/scripts/undetected.py))

### Profiling//Fingerprinting

A minus of undetected-chromedriver is, that it's __fingerprint depends__ on the __Operating-System__, which leads to:
- __inconsistency__ on different platforms
- websites might __analyze__ and detect, when a fingerprint is way to often used, since they usually are __unique__ (see [amiunique.org](https://amiunique.org/fp))

That's where Selenium-Profiles comes into play. It makes __specifying the fingerprint__ possible (==> emulation)

### Proxies

To make it as easy as possible, Selenium-Profiles allready includes support for authentificated proxies. This is quite important, since residential proxy-providers like [asocks](https://asocks.com/c/VBzkbI) usually require a password.

### Headless

Headless would be usefull for reducing computing-resources, but is __quite tricky__:
- `--headless` argument leaves out a lot of javascript objects ==> detection
- `--headless=chrome` seems to be better, but still isn't traceless.

__solution__
- __overwrite__ the missing __javascript objects__. That would also improve the fingerprinting-profiles. [bowserless](https://github.com/microlinkhq/browserless/tree/master/packages/goto/src/evasions) and [creep-js](https://github.com/abrahamjuliot/creepjs) would be a starting point.
- __unofficial workaround__. I can't tell a lot about it currently, since I didn't discover it, and the goal currently is, to keep it out of radar from sites like cloudfare.


### Target
The target is for this project, to make it as __complete__ as possible, as __computing-efficient__ as possible  and __pass [creep-js](https://abrahamjuliot.github.io/creepjs/)__ . It's the most advanced detection I've seen so far.

There are currently a lot of [todos](https://github.com/kaliiiiiiiiii/Selenium-Profiles#todo) and things to improve, but I currently don't have the capacity to carry on. But I'd __apprecheate contribution__ or __working together with undetected-chromedriver__. @ultrafunkamsterdam @sebdelsol

**Comment by alexreg** [comment]
@kaliiiiiiiiii What a great detailed reply... thank you very much! I can definitely see the reasoning behind this approach. Even if right now not many sites are (effectively) fingerprinting the browser/device, I'm sure more and more sites will start doing this in the future. And yes, good proxy support is very important too.

Anyway, I'm definitely up for trying out your library in the project I'm working on right now, so I can let you know how it goes and report bugs. Fingers crossed.

**Reply by alexreg** [reply]
P.S. I think you meant `--headless=new` is better but not traceless right? `--headless=chrome` should be the same as just `--headless`, i.e. the old method.

**Reply by ttlns** [reply]
> @kaliiiiiiiiii What a great detailed reply... thank you very much! I can definitely see the reasoning behind this approach. 

Thanks!
> Even if right now not many sites are (effectively) fingerprinting the browser/device, I'm sure more and more sites will start doing this in the future. And yes, good proxy support is very important too.

Yeah, but I assume that websites like cloudfare all ready have approaches.  For example detecting when there's a big peak in the statistics for a specific device.

> Anyway, I'm definitely up for trying out your library in the project I'm working on right now, so I can let you know how it goes and report bugs. Fingers crossed.

Great:) the newest version has some bugs (since proxy-support) regarding extensions. But I think the rest should be quite stable (except of the features I didn't ever use:) )

> P.S. I think you meant `--headless=new` is better but not traceless right? `--headless=chrome` should be the same as just `--headless`, i.e. the old method.

Mhmm no, I actually meant `--headless=chrome`.  Why do you mean it's the same?

**Reply by alexreg** [reply]
Sounds good.

Re headless mode, see [this Selenium blog post](https://www.selenium.dev/blog/2023/headless-is-going-away/) for example, which makes clear `--headless=chrome` (= `--headless`) is the old one and `--headless=new` is the new one.

**Comment by ultrafunkamsterdam** [comment]
None of this a is true . Dig way deeper....

**Reply by ttlns** [reply]
Regarding what do you mean? Headless? Or fingerprinting?
And if you say "dig way deeper", can you tell me what's wrong about?

**Comment by ttlns** [comment]
Closing as it seems to be resolved
